### DISCUSSION TITLE
Reproductibility

### DISCUSSION THREAD

**Discussion by tperchoc** [tancheng/CGRA-Mapper#31] [Q&A] [answered]
Hello,  
Thank you for your work, just
I’m having difficulties replicating the example FIR kernel mapping from `fir.cpp` to `kernel.bc` and achieving the expected simple DFG with II=1.  
Despite disabling optimizations, extracting only the `kernel` function, and adjusting `param.json`, I always get a more complex DFG and higher II.  
Could you please clarify the exact steps, compiler commands, and LLVM version you use to produce the `kernel.bc` shown in your documentation?  
I
Thank you 

What i get : 
![fir](https://github.com/user-attachments/assets/3d4bbf6e-faca-4d5a-ac9b-7b2e13d073e5)
What i should have :
![fir](https://github.com/user-attachments/assets/59333c33-9779-4acd-9df7-ad2908247abd)

I believe its comming from fir.cpp -> kernel.bc, i tried : 
clang-12 -O0 -emit-llvm -c fir.cpp -o kernel.bc
clang-12 -O0 -fno-inline -fno-unroll-loops -fno-vectorize -fno-slp-vectorize \
-emit-llvm -c fir.cpp -o kernel.bc
clang-12 -O0 -x c -std=c99 -fno-builtin -fno-inline -fno-unroll-loops \
-fno-vectorize -fno-slp-vectorize -emit-llvm -c fir.cpp -o kernel.bc

**Comment by tancheng** [comment]
Plz try -O3 for clang.

@MeowMJ plz help if O3 doesn't work. Thanks~!

**Comment by tperchoc** [comment]
![fir](https://github.com/user-attachments/assets/5b9c0ad1-6fd7-43c9-8c01-2eca38b213e4)
Thank you for responding so quickl
Okay with O3 i got this do you think its good ?

**Comment by tancheng** [comment, marked as answer]
The DFG is from unrolling the loop, we need to disable the unrolling. Did you use `clang-12 -O3 -fno-inline -fno-unroll-loops -fno-vectorize -fno-slp-vectorize -emit-llvm -c fir.cpp -o kernel.bc`?

**Reply by tperchoc** [reply]
I just used the command I found while searching in CGRA-Flow and it works, thank you very much
